src/x/net/http/errors.go

```diff
@@ -96,6 +98,8 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 	case error:
 		if xerrors.IsInvalidParams(v) {
 			w.WriteHeader(http.StatusBadRequest)
+		} else if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
So the correct error here is `context.DeadlineExceed` per go docs (and from testing).
From https://golang.org/pkg/context/: 
```
DeadlineExceeded is the error returned by Context.Err when the context's deadline passes.

var DeadlineExceeded error = deadlineExceededError{}
```